lib/finch/http1/conn.ex

```diff
@@ -108,36 +108,77 @@ defmodule Finch.Conn do
 
     start_time = Telemetry.start(:request, metadata, extra_measurements)
 
-    case HTTP1.request(conn.mint, req.method, full_path, req.headers, req.body) do
+    case HTTP1.request(conn.mint, req.method, full_path, req.headers, stream_or_body(req.body)) do
       {:ok, mint, ref} ->
-        Telemetry.stop(:request, start_time, metadata, extra_measurements)
-        start_time = Telemetry.start(:response, metadata, extra_measurements)
-
-        case receive_response([], acc, fun, mint, ref, receive_timeout) do
-          {:ok, mint, acc} ->
-            Telemetry.stop(:response, start_time, metadata, extra_measurements)
-            {:ok, %{conn | mint: mint}, acc}
+        case maybe_stream_request_body(mint, ref, req.body, receive_timeout) do
+          {:ok, mint} ->
+            Telemetry.stop(:request, start_time, metadata, extra_measurements)
+            track_response_telemetry(conn, metadata, extra_measurements, fn ->
+              receive_response([], acc, fun, mint, ref, receive_timeout)
+            end)
 
           {:error, mint, error} ->
-            metadata = Map.put(metadata, :error, error)
-            Telemetry.stop(:response, start_time, metadata, extra_measurements)
-            {:error, %{conn | mint: mint}, error}
+            handle_request_error(
```

Comment:
This will call `Telemetry.stop(:request, ...` but here we actually want to call stop for `:response`.

Reply:
Ah right, sorry, ignore me :)

Reply:
Never ;)